### PR TITLE
Compat update for String rename

### PR DESCRIPTION
**Pull request by StefanKarpinski**
There are still test failures but hopefully this is helpful.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -53,8 +53,8 @@ seekstart(b)
 r = Zlib.Reader(b)
 @test_throws ErrorException write(r, convert(UInt8, 20))
 for x in data
-    if typeof(x) == ASCIIString
-        @test x == ASCIIString(read(r, UInt8, length(x)))
+    if typeof(x) == String
```
Comment (yuyichao) [inline test/runtests.jl]: This change doesn't even make sense on 0.4

File: test/runtests.jl [hunk 2]
```diff
@@ -53,8 +53,8 @@ seekstart(b)
...
+        @test x == String(read(r, UInt8, length(x)))
```
Comment (yuyichao) [inline test/runtests.jl]: This is not supported on 0.4 either.

Comment (StefanKarpinski) [inline test/runtests.jl]: On 0.4 `Compat.String` is an alias for `UTF8String`.

Comment (yuyichao) [inline test/runtests.jl]: No it's not. It's `ByteString`.

Comment (yuyichao) [inline test/runtests.jl]: Ref https://github.com/JuliaLang/Compat.jl/pull/197#discussion_r62763761

**Comment by StefanKarpinski** [comment]
I had not kept up with that change to Compat, I was just trying to be helpful here. TBH, @yuyichao, you're being a bit unpleasant about this.

**Comment by yuyichao** [comment]
I'm happy to take advises on how to repeat exactly the same argument on ten different PR's (and a few of them containing changes that are clearly just a simple string replacement) without getting any ack in a 7 hours. I do appreciate your effort to fix the depwarn caused by your String renaming. However, if is unfortunate that since this is a merging of types, the renaming is the simplest part when fixing a package. Most of the time when fixing a package due to this change are spent on checking the expect usage and decide if they can be replaced with `String` or kept as `Compat.*`.

**Comment by StefanKarpinski** [comment]
A few of these PRs were actually helpful so it wasn't a total loss.

**Comment by yuyichao** [comment]
> A few of these PRs were actually helpful.

I agree, and I believe I also commented `LGTM` (or similar) on many of those.

I'm mainly annoyed by the few (those that I closed immediately) that are reverting fixes that are carefully made by others without looking at the commits that introduce them or on-going PRs. (and I do believe it should be a requirement to have a look at on-going PR's before/after making package PR's especially for those that doesn't have many open ones).

**Pull request closed without merging**